FundTrack.BLL/Concrete/ReportService.cs

```diff
@@ -101,5 +102,96 @@ public IEnumerable<String> GetImagesById(int finOpId)
                 throw new BusinessLogicException("Error while getting image path list from FinOPImages entities by finOpId.", ex);
             }
         }
+
+        public async Task<int> GetCountOfUsersDonationsReport(int orgId, DateTime dateFrom, DateTime dateTo)
+        {
+            var t = await GetUserDonations(orgId, dateFrom, dateTo);

```

Comment:
this opertation retreives all records from the database, then EF proceses them and then you just abandon them and take a count of records. Too much work is done behihd the scenes, consider executing query that just executes SELECT COUNT.